lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp

```diff
@@ -86,5 +93,81 @@ void StoreOp::build(OpBuilder &b, OperationState &state, Value ptr, Value value,
   build(b, state, ptr, value, dynamicDims, b.getDenseI64ArrayAttr(staticDims));
 }
 
+LogicalResult GetStructuredStateOp::verify() {
+  return success();
+  auto expectedOffsetAndStrideTypes =
+      getOffsetAndStrideTypes(getContext(), getStructuredPtr().getType());
+
+  if (!expectedOffsetAndStrideTypes.has_value()) {
+    return failure();
+  }
+
+  auto [expectedOffsetTypes, expectedStrideTypes] =
+      *expectedOffsetAndStrideTypes;
+
+  return success(expectedOffsetTypes.size() == getOffsets().size() &&
```

Comment:
Does `llvm::equal` not override for the `.size()` type?

Reply:
yeah it is a wrapper for std::equal which doesn't do bound check